File: tests/corelib/string_trim2_test.dart
// Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import "package:expect/expect.dart";

const WHITESPACE = const [
  0x09,
  0x0A,
  0x0B,
  0x0C,
  0x0D,
  0x20,
  0x85,
  0xA0,
  0x1680,
  0x2000,
  0x2001,
  0x2002,
  0x2003,
  0x2004,
  0x2005,
  0x2006,
  0x2007,
  0x2008,
  0x2009,
  0x200A,
  0x202F,
  0x205F,
  0x3000,
  0x2028,
  0x2029,
  0xFEFF,
];

main() {
  for (var ws in WHITESPACE) {
    var name = ws.toRadixString(16);
    var c = new String.fromCharCode(ws);
    Expect.equals("", c.trim(), "$name");
    Expect.equals("a", ("a" + c).trim(), "a-$name");
    Expect.equals("a", (c + "a").trim(), "$name-a");
    Expect.equals("a", (c + c + "a" + c + c).trim(), "$name around");
    Expect.equals(
        "a" + c + "a", (c + c + "a" + c + "a" + c + c).trim(), "$name many");
  }
  Expect.equals("", new String.fromCharCodes(WHITESPACE).trim(), "ALL");
}

File: pkg/dev_compiler/test/codegen_expected/corelib/string_trim2_test.js
dart_library.library('corelib/string_trim2_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__string_trim2_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const string_trim2_test = Object.create(null);
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  string_trim2_test.WHITESPACE = dart.constList([9, 10, 11, 12, 13, 32, 133, 160, 5760, 6158, 8192, 8193, 8194, 8195, 8196, 8197, 8198, 8199, 8200, 8201, 8202, 8239, 8287, 12288, 8232, 8233, 65279], core.int);
  string_trim2_test.main = function() {
    for (let ws of string_trim2_test.WHITESPACE) {
      let name = ws[dartx.toRadixString](16);
      let c = core.String.fromCharCode(ws);
      expect$.Expect.equals("", c[dartx.trim](), dart.str`${name}`);
      expect$.Expect.equals("a", ("a" + c)[dartx.trim](), dart.str`a-${name}`);
      expect$.Expect.equals("a", (c + "a")[dartx.trim](), dart.str`${name}-a`);
      expect$.Expect.equals("a", (c + c + "a" + c + c)[dartx.trim](), dart.str`${name} around`);
      expect$.Expect.equals("a" + c + "a", (c + c + "a" + c + "a" + c + c)[dartx.trim](), dart.str`${name} many`);
    }
    expect$.Expect.equals("", core.String.fromCharCodes(string_trim2_test.WHITESPACE)[dartx.trim](), "ALL");
  };
  dart.fn(string_trim2_test.main, VoidTodynamic());
  // Exports:
  exports.string_trim2_test = string_trim2_test;
});
